### DISCUSSION TITLE
Using pkg for node and react front end

### DISCUSSION THREAD

**Discussion by xorCoding** [yao-pkg/pkg#59] [Q&A] [answered]
Hi, i'm trying to use pkg to create a node web server that uses react for front-end. I created the react app with "create-react-app".
Here's my "server.js" file that i compile with pkg:

```
const path = require("path");
const express = require("express");
const app = express(); // create express app

// add middlewares
app.use(express.static(path.join(__dirname, "./client", "build")));
app.use(express.static("public"));

app.use((req, res, next) => {
  res.sendFile(path.join(__dirname, "..", "build", "index.html"));
});

// start express server on port 5000
app.listen(5000, () => {
  console.log("server started on port 5000");
});
```

The .exe gets created correctly, and the server also starts when i run it, but i get this error on the browser:

> Error: File or directory 'C:\**\build\index.html' was not included into executable at compilation stage. Please recompile adding it as asset or script.

Here's the full error:
![immagine](https://github.com/yao-pkg/pkg/assets/107063555/b9b7ca62-572b-4caf-a7f1-58ded885fabf)

Is it possible to do what I'm trying? If yes, how can I fix it?

**Comment by robertsLando** [comment, marked as answer]
You have to add /build folder to assets

**Reply by xorCoding** [reply]
Hi, I tried that but I'm sure that i'm making some mistakes.
This is my folder:
![immagine](https://github.com/yao-pkg/pkg/assets/107063555/f04ea69f-043d-4703-9acb-be3e14220338)
And this is my package.json:
![immagine](https://github.com/yao-pkg/pkg/assets/107063555/08df1692-390f-49ae-bb19-2302eca204ab)

What should I change in order to make it work? Thanks in advance

**Reply by xorCoding** [reply]
Nevermind, I fixed it lmao
